R/PredictionDataSurv.R

```diff
@@ -132,7 +132,13 @@ filter_prediction_data.PredictionDataSurv = function(pdata, row_ids, ...) {
     distr = pdata$distr
 
     if (testDistribution(distr)) { # distribution
-      pdata$distr = distr[keep]
+      ok = inherits(distr, c("VectorDistribution", "Matdist", "Arrdist")) &&
+           length(keep) > 1 # edge case: Arrdist(1xYxZ) and keep = FALSE
+      if (ok) {
+        pdata$distr = distr[keep] # we can subset row/samples like this
+      } else {
+        pdata$distr = base::switch(keep, distr) # one distribution only
+      }
```

Comment:
@bblodfon can you tell me what's happening here?

Reply:
I was trying to filter `Distribution` objects (like the prediction object from `surv.parametric`) since I noticed we don't do any filtering on those and I needed on the tests to compare RCLL old and new code kind-of. The above is the result of testing some edge cases. 

In general you can subset with `distr[keep]` when you have a `Matdist`, `Arrdist` or an `VectorDistribution` if you have two or more observations. The problem starts when 1 observation is left. `distr[keep]` with a `VectorDistribution` leaves a `Distribution` (e.g. `WeibullAFT`-something in `surv.parametric`), a `WeightedDiscrete()` for `Matdist`, and an `Arrdist` of dim `1x...` (it doesn't degenerate to a `Matdist`). And that's all fine, we still use `pdata$distr = distr[keep]` to do this. But then if you redo some more filtering (on that same specific row index or another to go to 0 observations), `keep` will be either `TRUE` or `FALSE`, and most of the above are not subsettable or don't work at all:

Reply:
![image](https://github.com/mlr-org/mlr3proba/assets/10909555/a8d5a291-f21f-4e68-845a-a7c6a9aa0d4b)
![arrdist_logical_subsetting](https://github.com/mlr-org/mlr3proba/assets/10909555/1bdc7eb6-af05-4bdd-92f0-13fcdfdd16cc)

Reply:
Thanks for asking this - based on the above, we might want to make an `Arrdist` to a `Matdist` when only one observation is left by design in `distr6`?

Reply:
@RaphaelS1 Do you have an opinion on this?